pyNN/common/procedural_api.py

```diff
@@ -74,7 +74,11 @@ def record(variables, source, filename, sampling_interval=None, annotations=None
         # would actually like to be able to record to an array and choose later
         # whether to write to a file.
         if not isinstance(source, (BasePopulation, Assembly)):
-            source = source.parent
+            if isinstance(source, (IDMixin)):
+                source = source.as_view()
+            else:
+                # not sure when this would be used; retaining earlier code
+                source = source.parent
```

Comment:
@apdavison : do you see any scenario where this would be used? should we remove it?